src/python/gudhi/simplex_tree.pyx

```diff
@@ -195,6 +195,49 @@ cdef class SimplexTree:
         """
         return self.get_ptr().insert(simplex, <double>filtration)
 
+    def insert_edges_from_array(self, filtrations, double max_filtration=numpy.inf):
+        """Inserts edges in an empty complex. The vertices are numbered from 0 to n-1, and the filtration values are
+        encoded in the array, with the diagonal representing the vertices. It is the caller's responsibility to
+        ensure that this defines a filtration, which can be achieved with either::
+
+            filtrations[np.diag_indices_from(filtrations)] = filtrations.min(1)
+
+        or::
+
+            diag = filtrations.diagonal()
+            filtrations = np.fmax(np.fmax(filtrations, diag[:, None]), diag[None, :])
+
+        :param filtrations: the filtration values of the vertices and edges to insert. The matrix is assumed to be symmetric.
+        :type filtrations: numpy.ndarray of shape (n,n)
+        :param max_filtration: only insert vertices and edges with filtration values no larger than max_filtration
+        :type max_filtration: float
+        """
+        # TODO: document which half of the matrix is actually read?
```

Comment:
I can affirm it is the upper part of the matrix that is read, and I confirm it would be nice to be documented.

Reply:
My test to see if it is the same result as [rips distance matrix](https://gudhi.inria.fr/python/latest/rips_complex_user.html#example-from-a-distance-matrix):
```python
import numpy as np
import gudhi

a = np.array([[0., 6.0827625303, 5.8309518948, 9.4339811321, 13.0384048104, 18.0277563773, 17.88854382], [0., 0., 6.7082039325, 6.3245553203, 15.6524758425, 19.6468827044, 17.1172427686], [0., 0., 0., 5, 8.94427191, 13.152946438, 12.0830459736], [0., 0., 0., 0., 12.0415945788, 14.7648230602, 11], [0., 0., 0., 0., 0., 5.3851648071, 9.4868329805], [0., 0., 0., 0., 0., 0., 7.2801098893], [0., 0., 0., 0., 0., 0., 0.]])
st.insert_edges_from_array(a)
st = gudhi.SimplexTree()
st.insert_edges_from_array(a, max_filtration = 12.)

print(list(st.get_filtration()))
print("num simplices = ", st.num_simplices(), "num vertices = ", st.num_vertices())
```

It is strange to have one functionality with the upper and the other one with the lower, but if it is documented, I will let you decide.

Reply:
I think it would be easy to switch (at worst, just transpose the matrix, which takes constant time), I don't care which half is read.
I should try to measure if restricting to numpy arrays really brings a measurable gain, or if accessing it as a sequence of sequences (or iterables) is good enough.